orthoinference/Jenkinsfile

```diff
@@ -0,0 +1,229 @@
+import groovy.json.JsonSlurper
+
+pipeline{
+	agent any
+	
+	stages{
+		stage('Check if upstream builds succeeded'){
+			steps{
+				script{
+					def orthopairsStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/Orthopairs/lastBuild/api/json"
+					def orthopairsStatusJson = new JsonSlurper().parseText(orthopairsStatusUrl.getContent())
+					if(orthopairsStatusJson['result'] != "SUCCESS"){
+						error("Most recent Orthopairs build status: " + orthopairsStatusJson['result'])
+					}
+			    	}
```

Comment:
If possible, a utility function for checking a previous build would be great with parameters of what to check and any error message text to append to a common message.

Reply:
Done